services/command-router/src/main/java/org/eclipse/hono/commandrouter/impl/CommandRouterServiceImpl.java

```diff
@@ -237,16 +268,65 @@ private Duration getSanitizedLifespan(final Duration lifespan) {
 
     @Override
     public Future<CommandRouterResult> unregisterCommandConsumer(final String tenantId, final String deviceId,
-            final String adapterInstanceId, final Span span) {
+            final boolean sendEvent, final String adapterInstanceId, final Span span) {
 
         return deviceConnectionInfo
                 .removeCommandHandlingAdapterInstance(tenantId, deviceId, adapterInstanceId, span)
-                .recover(thr -> {
+                .onFailure(thr -> {
                     if (ServiceInvocationException.extractStatusCode(thr) != HttpURLConnection.HTTP_PRECON_FAILED) {
-                        LOG.info("error removing command handling adapter instance [tenant: {}, device: {}]", tenantId, deviceId, thr);
+                        LOG.info("error removing command handling adapter instance [tenant: {}, device: {}]", tenantId,
+                                deviceId, thr);
                     }
-                    return Future.failedFuture(thr);
                 })
+                .compose(v -> {
+                    if (sendEvent) {
+                        return tenantClient.get(tenantId, span.context()).compose(
+                                tenantObject -> sendDisconnectedTtdEvent(tenantObject, deviceId, adapterInstanceId,
```

Comment:
If `sendEvent` is true, the event should also be sent if `removeCommandHandlingAdapterInstance` failed. Sending should only be skipped if there was a `HTTP_PRECON_FAILED` error status on `removeCommandHandlingAdapterInstance`.
See the logic in the old `AbstractVertxBasedMqttProtocolAdapter.onCommandSubscriptionRemoved()` method.

Reply:
Right, I have missed that flow. Will fix that.